### ISSUE TITLE
Feature: Possibility to change indentation of bullet points with tab key

### ISSUE THREAD

**Issue by yennor** [tomboy-notes/tomboy-ng#323]
I'm used from a lot of application, that I can change the indentation (the level) of bullet points using tab and shift-tab. 
I could get used to use tab-<left arrow> and tab-<right arrow>. \
But it still would be nice if i'd be possible using the tab key.

**Comment by davidbannon** [comment]
OK, you have, apparently, appointed yourself as an Alpha-Tester !

See if you prefer this. I have to say, I was uncomfortable with that Indent model anyway, it was flagged as experimental in last release and, I thing, thats where it says.

So, in attached, tab and shift-tab increase and decrease bullet levels (along with Alt Left/Right still).

There is no intent ability, notes containing an indent will load with that text flush left. 

Your feedback will be appreciated.

Davo

[tomboy-ng.zip](https://github.com/user-attachments/files/18210007/tomboy-ng.zip)

**Comment by aguador** [comment]
Couldn't avoid the testing fun! I did a quick test with newly created notes and one old note: both tab/shift-tab and Alt-Left/Alt-Right changed bullets as advertised. The test version removed a tabbed indent I created in 0.41. When later opened in 0.41, the indent was gone. So the the tab is removed/irreversible.

Question to ponder: If this new behaviour is instituted, whould users be polled or warned about it? I can see folks with notes which contain tabs (not my case) being less than happy.

Note: Please see my note on Issues 322 (access violation) as that problem may be Qt-related.

**Comment by aguador** [comment]
Oops, sorry, my bad . I did not see the part about the cursor being eol. I was able to replicte the bug.

**Comment by davidbannon** [comment]
Yes, [aguador](https://github.com/aguador), definitely irreversible. Open an old note containing a Tab (as constructed by version 0.40 or 0.41) and its silently (and permanently removed. That is, of course, only in this experimatal version posted here. 'We' need decide if it is a solution of another backwater ...

I have managed this tab indent badly. It was an experimental feature in one of the 0.39+ betas, was clearly described as experimental then. But I left it in there in the 0.40 and then 0.41 release unchanged and, perhaps (?) unheralded. My bad !

Then yennor discovered the crashing bug where shift-tab, when used at the end of a line does bad things. I must rush out a fix for urgently, while the note is safe, contented added up to 10 seconds previously is not. And data loss is unacceptable.

I am constrained by how KMemo handles things, to get that tab function I worked another wrapper around KMemo like I had to do with bullets, not pretty. And, evidence is, buggy. The intention was, always, just one indent, remove indent with Shift-Tab or Backspace at start of line. Honestly, not a particularly useful feature IMHO. So, again, IMHO, easy thing to do is to remove it, I don't see any value.

But, then, https://github.com/tomboy-notes/tomboy-ng/issues/324 arrives !

So, summary, KMemo does not handle tabs (well). Older versions of -ng just inserted 8 spaces, made sense at beginning of para. Ver 0.40 and 0.41 has a buggy insert function. Need to to do better.

Davo

**Comment by davidbannon** [comment]
Please see #322 for an update.

**Comment by davidbannon** [comment]
I have made a Beta Release of tomboy-ng that, I think, addresses the issue you have reported here. You might like to try it out and see if it solves your problem.
Thanks for your input.
David

**Comment by davidbannon** [comment]
The new release v0.42 of tomboy-ng, https://github.com/tomboy-notes/tomboy-ng/wiki/Download_Release resolves this issue. 
Thank you for your report.
Closing.
Davo
